tests/ga/test_remoteaccess.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2014 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+from tests.tools import *
+from azurelinuxagent.common.protocol.wire import *
+from azurelinuxagent.common.osutil import get_osutil
+
+class TestRemoteAccess(AgentTestCase):
+    def test_parse_remote_access(self):
+        data_str = load_data('wire/remote_access_single_account.xml')
+        remoteAccess = RemoteAccess(data_str)
+        self.assertNotEquals(None, remoteAccess)
+        self.assertEquals(1, remoteAccess.Incarnation)
+        self.assertEquals(1, len(remoteAccess.Users))
+        self.assertEquals("testAccount", remoteAccess.Users[0].Name)
+        self.assertEquals("encryptedPasswordString", remoteAccess.Users[0].EncryptedPassword)
+        self.assertEquals("2019-01-01", remoteAccess.Users[0].Expiration)
+        #self.assertEquals(2, len(remoteAccess.Users[0].Groups))
```

Comment:
You didn't implement Groups, so I would delete the code.

Reply:
Why?  Groups are not required?  We don't provide users a method to specify groups in our UI.

Reply:
You cannot create a Linux user without specifying the primary group to which that user belongs; it's simply impossible. That probably means your UI is insufficient to the task. If you plan to support adding users with anything other than "ordinary user" capabilities, you need to support supplemental group membership; for example, if your requirements include things like "create a user who can access system logs" then you need to make that user a member of the "syslog" supplemental group.

Reply:
This works in Ubuntu as-is... I'll look to see what I'm missing though but I don't remember seeing anything that added the user to a group (I actually initially wrote the code to add users to groups but removed it)  

Our requirements for Linux though do not include low privileged rights.  Sudo privileges is all that's needed.